third_party/2and3/google/protobuf/internal/extension_dict.pyi

```diff
@@ -10,13 +10,27 @@ class _ExtensionFieldDescriptor(FieldDescriptor, Generic[_ContainerMessageT, _Ex
 
 class _ExtensionDict(Generic[_ContainerMessageT]):
     def __init__(self, extended_message: _ContainerMessageT) -> None: ...
+    # Dummy fallback overloads with FieldDescriptor are for backward compatibility with
+    # mypy-protobuf <= 1.23. We can drop them a few months after 1.24 releases.
+    @overload
     def __getitem__(
         self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT]
     ) -> _ExtenderMessageT: ...
+    @overload
+    def __getitem__(self, extension_handle: FieldDescriptor) -> Any: ...
+    @overload
     def __setitem__(
         self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT], value: _ExtenderMessageT
     ) -> None: ...
-    def __delitem__(self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT]) -> None: ...
-    def __contains__(self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, _ExtenderMessageT]) -> bool: ...
+    @overload
+    def __setitem__(self, extension_handle: FieldDescriptor, value: Any) -> None: ...
+    @overload
+    def __delitem__(self, extension_handle: _ExtensionFieldDescriptor[_ContainerMessageT, Any]) -> None: ...
```

Comment:
What about using `_ExtenderMessageT` instead of `Any` here?

Reply:
IMO it is kind of misleading, since `_ExtenderMessageT` is not fixed by anything (`_ExtensionDict` is only generic in one type argument, IIUC because it is polymorphic).

Reply:
Ok, so this would only potentially remove some warnings about `Any` types.

Reply:
Check out this comment thread from the original diff
https://github.com/python/typeshed/pull/4619/files#r500533547

Reply:
Actually, maybe you are right, these warnings could be annoying. Another way may be to use `object` and make `_ExtensionFieldDescriptor` covariant in the second argument, but just putting that type variable back is easier.